packages/py-moose-lib/moose_lib/streaming/streaming_function_runner.py

```
@@ -494,6 +494,27 @@ def process_messages():
+                # poll(0) triggers group join without blocking
+                consumer.poll(timeout_ms=0)
+                assignment = consumer.assignment()
+                if assignment:
+                    log(
+                        f"Consumer ready with {len(assignment)} partition(s): {assignment}"
+                    )
+                    break
+                if time.time() - start_time > max_wait_seconds:
+                    raise RuntimeError(
+                        f"Consumer failed to get partition assignment within {max_wait_seconds}s"
+                    )
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)